File: leetcode/이진검색트리의 기본 성질 530. Minimum Absolute Difference in BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private boolean isInit;
    private int prev;
    private int minDiff;

    public int getMinimumDifference(TreeNode root) {
        isInit = false;
        minDiff = Integer.MAX_VALUE;
        inorder(root);
        return minDiff;
    }

    private void inorder(TreeNode node) {
        if (node == null) return;

        inorder(node.left);
        if (!isInit) {
            isInit = true;
        } else {
            minDiff = Math.min(minDiff, node.val - prev);
        }
        prev = node.val;
        inorder(node.right);
    }
}

File: leetcode/이진검색트리의 기본 성질 530. Minimum Absolute Difference in BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
import math


class Solution:
    def getMinimumDifference(self, root: TreeNode) -> int:
        self.minDiff = math.inf
        self.isInit = False
        self.prev = None
        self.inorder(root)
        return self.minDiff

    def inorder(self, root):
        if root == None:
            return

        self.inorder(root.left)
        if not self.isInit:
            self.isInit = True
        else:
            self.minDiff = min(self.minDiff, root.val - self.prev)

        self.prev = root.val
        self.inorder(root.right)